### PR TITLE
test: Validation Suite for Stock Ledger Invariant Check – Columns, Execution, and Reposting

### PR DESCRIPTION
**Pull request by hrishikesh8848**
### Title: 
Add Test Cases for Stock Ledger Invariant Check Report

### Test Summary:
This test suite adds automated test cases for the Stock Ledger Invariant Check report in ERPNext. It ensures that the report executes properly, handles FIFO queue edge cases, and accurately creates reposting entries through both dictionary and JSON string inputs. It also validates that the report's column definitions are correctly structured.

### Scenarios Covered
Input Validations
test_report_execution_T_SLIC_001
Verifies that the report runs successfully with basic filters (item_code, warehouse) and returns a valid structure: list of columns and non-empty data with key fields like fifo_valuation_rate.

test_report_columns_T_SLIC_005
Checks that the report's column definitions include key fields like valuation_diff, ensuring consistency with expected report schema.

Edge Cases
test_empty_fifo_queue_T_SLIC_002
Simulates FIFO queue consumption by issuing a stock outflow and ensures the report identifies it properly, confirming the presence of consumption_rate in results.

API Behavior & Data Consistency
test_create_reposting_entries_T_SLIC_003
Tests the backend utility to programmatically create reposting entries using structured Python dict input and verifies that no exceptions occur during execution.

test_create_reposting_entries_with_string_input_T_SLIC_004
Verifies that the same reposting logic can handle stringified JSON input (as might be passed from a frontend) without errors.

### Technology:

Framework: unittest with FrappeTestCase utilities

### Linked Feature/Bug:

None

### CODE REVIEW

File: erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+from erpnext.stock.doctype.stock_entry.test_stock_entry import make_stock_entry
+from erpnext.stock.report.stock_ledger_invariant_check import stock_ledger_invariant_check as report
+
+
+class TestStockLedgerInvariantCheck(unittest.TestCase):
+	def setUp(self):
+		frappe.set_user("Administrator")
+
+		self.item = create_item(item_code="_Test Item", is_stock_item=1, valuation_rate=100)
+		self.warehouse = "_Test Warehouse - _TC"
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py]: Create or get warehouse before using it

File: erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+				"posting_date": se_doc.posting_date,
+				"posting_time": se_doc.posting_time,
+				"voucher_type": se_doc.doctype,
+				"voucher_no": se_doc.name,
+				"company": se_doc.company,
+				"qty": se_doc.items[0].qty,
+				"valuation_rate": se_doc.items[0].basic_rate,
+			}
+		]
+		report.create_reposting_entries(rows)
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py:88]: what are we testing here?

File: erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+				"posting_date": se_doc.posting_date.isoformat(),
+				"posting_time": se_doc.posting_time.isoformat(),
+				"voucher_type": se_doc.doctype,
+				"voucher_no": se_doc.name,
+				"company": se_doc.company,
+				"qty": float(se_doc.items[0].qty),
+				"valuation_rate": float(se_doc.items[0].basic_rate),
+			}
+		]
+		report.create_reposting_entries(json.dumps(rows))
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py:133]: what are we testing here

**Review by aasif-patel — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	def test_create_reposting_entries_T_SLIC_003(self):
+		se = make_stock_entry(
+			item=self.item.name,
+			qty=5,
+			rate=100,
+			to_warehouse=self.warehouse,
+			posting_date=nowdate(),
+			posting_time=nowtime(),
+		)
+		se_doc = frappe.get_doc("Stock Entry", se.name)
```
Comment (aasif-patel) [inline erpnext/stock/report/stock_ledger_invariant_check/test_stock_ledger_invariant_check.py:65]: if we already have se as a  StockEntry object then why are we getting the doc again?

**Review by aasif-patel — CHANGES_REQUESTED** [review]
